x-pack/plugin/async-search/src/internalClusterTest/java/org/elasticsearch/xpack/search/CrossClusterAsyncSearchIT.java

```diff
@@ -413,40 +423,44 @@ public void testClusterDetailsAfterCCSWithFailuresOnAllShards() throws Exception
 
         boolean minimizeRoundtrips = TransportSearchAction.shouldMinimizeRoundtrips(request.getSearchRequest());
 
-        AsyncSearchResponse response = submitAsyncSearch(request);
-        assertNotNull(response.getSearchResponse());
-
-        waitForSearchTasksToFinish();
-
-        {
-            AsyncSearchResponse finishedResponse = getAsyncSearch(response.getId());
-            assertTrue(finishedResponse.isPartial());
+        final AsyncSearchResponse response = submitAsyncSearch(request);
+        try {
+            assertNotNull(response.getSearchResponse());
+            waitForSearchTasksToFinish();
+            final AsyncSearchResponse finishedResponse = getAsyncSearch(response.getId());
+            try {
```

Comment:
This double nesting of try/finally blocks seems unnecessary - I don't think you've had to do it in other tests. What if you cache the response.getId() like elsewhere and then `response.decRef` before you start the second try/finally?

Reply:
++ indeed, that was just me getting tired I think. Much better this way, thanks for spotting!